src/include/network/lan/complex.rb

```diff
@@ -393,14 +446,23 @@ def handleOverview(_key, event)
           # As it could easily happen that all interfaces are set to "no" (and
           # default route is unrecheable in such case) this explicite setup was
           # added.
+          # FIXME: not implemented in network-ng
           LanItems.set_default_route = true
 
           return :add
         when :edit
-          if LanItems.IsCurrentConfigured
-            LanItems.SetItem
-
-            if LanItems.startmode == "managed"
+          iface = config.interfaces.find(iface_name)
+          # FIXME: turn following lines into something like builder.load(iface) ?
+          @builder.name = iface.name
+          @builder.type = iface.type
+          if iface.configured
+            @builder.load_sysconfig(iface.config)
+            @builder.load_s390_config(LanItems.s390_ReadQethConfig(iface.hardware.name)) if iface.hardware
```

Comment:
I hope that it should be hidden in that generic load.not job of UI to handle different internals for different arch. What is its job if there is specific option for s390, then we need to have specialized UI, but not for loading on s390.

Reply:
Whole Add / Edit / ... workflow handling needs some polishing as it is somehow spread over the code. We currently need to isolate what is needed to keep ... so thanks for any hint ;-)